app/frontend/src/components/ches/UploadContexts.vue

```diff
@@ -0,0 +1,385 @@
+
+        const headers = this.sanitizeHeaders(data[0]);
+        excel.headers = [headers];
+        // actual data is rows 2 onward
+        excel.data = data.slice(1);
+
+        let contexts = [];
+
+        excel.data.forEach((d) => {
+          let r = {
+            to: [],
+            cc: [],
+            bcc: [],
+            tag: undefined,
+            delayTS: undefined,
+            delayTSDisplay: undefined,
+            context: {},
+          };
+          let fields = excel.cols;
+          fields.forEach((f) => {
+            let fieldName = excel.headers[0][f.key];
+            switch (fieldName.toLowerCase()) {
+              case 'to':
+                r.to = Utils.getAddresses(d[f.key]);
+                break;
+              case 'cc':
+                r.cc = Utils.getAddresses(d[f.key]);
+                break;
+              case 'bcc':
+                r.bcc = Utils.getAddresses(d[f.key]);
+                break;
+              case 'tag':
+                r.tag = d[f.key];
+                break;
+              case 'delayts':
+                r.delayTS = this.parseDelayTs(d[f.key]);
+                // for display in the table, format it nicely (not unix milliseconds...)
+                d[f.key] = r.delayTS
+                  ? moment(d[f.key]).format(Formats.CONTEXTS_TS_FORMAT)
+                  : undefined;
+                break;
+              default:
+                this.handleDefault(r.context, d, fieldName, f.key);
+                // for display in the table, return the value from the context (altered for date or timestamp)
+                d[f.key] = r.context[fieldName];
+            }
+          });
+          if (this.validateContext(r)) contexts.push(r);
+        });
+
+        let formContexts = JSON.stringify(contexts, null, 2);
+        const contextVariables = this.contextsToVariables(formContexts);
+
+        cb({
+          excel: excel,
+          contexts: formContexts,
+          contextVariables: contextVariables,
+        });
+      };
+
+      if (rABS) reader.readAsBinaryString(file);
+      else reader.readAsArrayBuffer(file);
+    },
+
+    // add fields to store
+    addContextsToStore(data) {
+      let { excel, contexts, contextVariables } = data;
+      this.excel = excel;
+      this.contexts = contexts;
+      this.contextsType = 'xlsx';
+      this.contextVariables = contextVariables;
+    },
+
+    make_cols(refstr) {
+      let o = [],
+        C = XLSX.utils.decode_range(refstr).e.c + 1;
+      for (var i = 0; i < C; ++i)
+        o[i] = { name: XLSX.utils.encode_col(i), key: i };
+      return o;
+    },
+
+    sanitizeHeaders(headers) {
+      let result = [];
+      headers.forEach((c) => {
+        let h = this.makeHeaderUnique(result, Utils.toCamelCase(c));
+        result.push(h);
+      });
+      return result;
+    },
+
+    makeHeaderUnique(existing, original, val = original, count = 0) {
+      if ('' === original) {
```

Comment:
By standard JS convention, comparisons should always be done such that the element that can be variable should be first, and the constant value second.
```js
if (original === '') { ... }
```